design-proposal.md

```diff
@@ -0,0 +1,66 @@
+# Proposal: Design of the rkt + Kubernetes CRI
+
+## Abstract
+
+This proposal can be viewed as more a discussion than an actual proposal, because we still haven't reached a consensus on the architecture.
+
+## Backgroud
+
+As container runtime interface (CRI) proposals and original stub codes are already merged into Kubernetes, the actual work has already begun.
+All existing and future container runtimes are expected to conform the interface and provides feature parities.
+
+## Design(s)
+
+There are three different designs:
+
+- `Kubelet` --_gPRC calls_--> `rktlet service` --_invoke 'rkt' binary, or use some rkt library_--> result
```

Comment:
We just discussed solution 1. with @lucab OOB: This makes the existing rkt `api-service` obsolete. If we go with this direction we could get rid of `api-service` in the rkt source tree at all and host it here instead, making it a viable "initial commit" for the rktlet service.